// @azure/cognitiveservices-luis-authoring/esm/operations/examples.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/examplesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Examples. */
var Examples = /** @class */ (function () {
    /**
     * Create a Examples.
     * @param {LUISAuthoringClientContext} client Reference to the service client.
     */
    function Examples(client) {
        this.client = client;
    }
    Examples.prototype.add = function (appId, versionId, exampleLabelObject, options, callback) {
        return this.client.sendOperationRequest({
            appId: appId,
            versionId: versionId,
            exampleLabelObject: exampleLabelObject,
            options: options
        }, addOperationSpec, callback);
    };
    Examples.prototype.batch = function (appId, versionId, exampleLabelObjectArray, options, callback) {
        return this.client.sendOperationRequest({
            appId: appId,
            versionId: versionId,
            exampleLabelObjectArray: exampleLabelObjectArray,
            options: options
        }, batchOperationSpec, callback);
    };
    Examples.prototype.list = function (appId, versionId, options, callback) {
        return this.client.sendOperationRequest({
            appId: appId,
            versionId: versionId,
            options: options
        }, listOperationSpec, callback);
    };
    Examples.prototype.deleteMethod = function (appId, versionId, exampleId, options, callback) {
        return this.client.sendOperationRequest({
            appId: appId,
            versionId: versionId,
            exampleId: exampleId,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    return Examples;
}());
export { Examples };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var addOperationSpec = {
    httpMethod: "POST",
    path: "apps/{appId}/versions/{versionId}/example",
    urlParameters: [
        Parameters.endpoint,
        Parameters.appId,
        Parameters.versionId0
    ],
    queryParameters: [
        Parameters.enableNestedChildren
    ],
    requestBody: {
        parameterPath: "exampleLabelObject",
        mapper: __assign(__assign({}, Mappers.ExampleLabelObject), { required: true })
    },
    responses: {
        201: {
            bodyMapper: Mappers.LabelExampleResponse
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var batchOperationSpec = {
    httpMethod: "POST",
    path: "apps/{appId}/versions/{versionId}/examples",
    urlParameters: [
        Parameters.endpoint,
        Parameters.appId,
        Parameters.versionId0
    ],
    queryParameters: [
        Parameters.enableNestedChildren
    ],
    requestBody: {
        parameterPath: "exampleLabelObjectArray",
        mapper: {
            required: true,
            serializedName: "exampleLabelObjectArray",
            type: {
                name: "Sequence",
                element: {
                    type: {
                        name: "Composite",
                        className: "ExampleLabelObject"
                    }
                }
            }
        }
    },
    responses: {
        201: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "BatchLabelExample"
                        }
                    }
                }
            }
        },
        207: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "BatchLabelExample"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "apps/{appId}/versions/{versionId}/examples",
    urlParameters: [
        Parameters.endpoint,
        Parameters.appId,
        Parameters.versionId0
    ],
    queryParameters: [
        Parameters.skip,
        Parameters.take,
        Parameters.enableNestedChildren
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "LabeledUtterance"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "apps/{appId}/versions/{versionId}/examples/{exampleId}",
    urlParameters: [
        Parameters.endpoint,
        Parameters.appId,
        Parameters.versionId0,
        Parameters.exampleId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OperationStatus
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
